recipe/meta.yaml

```diff
@@ -1,63 +1,73 @@
 {% set name = "streamlit-extras" %}
-{% set version = "0.2.7" %}
+{% set version = "0.4.0" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
-  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/streamlit_extras-{{ version }}.tar.gz
-  sha256: 52a8cb3a4394735ee873c92c95fabf56d84cc2400ed0617bbc8b8259c0560276
+  url: https://github.com/arnaudmiribel/{{ name }}/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: d6e863ab86d251bfb8ffc2abcfcd0fc99583fdc0ba00f4936db51fa0a23766d5
   patches:
-    # entrypoints is a package in maintenance mode that seems to be implicitly used by streamlit-extras. It seems to be 
-    # not really needed. Rather than adding entrypoints to the runtime dependencies, it was decided to remove it from the implementation.
-    - patches/0001-remove-the-usage-of-entrypoints.patch
+  - patch/0001-updating-htbuilder-pinning.patch
 
 build:
   script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
-  number: 1
+  number: 0
   # s390x is missing streamlit
   skip: true  # [py<38 or s390x]
 
 
+
 requirements:
   build:
-    - patch  # [not win]
-    - m2-patch  # [win]
+    - patch       # [not win]
+    - m2-patch    # [win]
   host:
-    - python
+    - python !=3.9.7
     - poetry-core
     - pip
     - setuptools
     - wheel
   run:
-    - python <3.9.7,>3.9.7,<4.0
+    - python !=3.9.7
+    - entrypoints >=0.4
     - streamlit >=1.0.0
     - st-annotated-text >=3.0.0
     - streamlit-embedcode >=0.1.2
     - streamlit-keyup >=0.1.9
     - protobuf !=3.20.2
     - streamlit-camera-input-live >=0.2.0
-    - streamlit-vertical-slider >=1.0.2
+    - streamlit-vertical-slider >=2.5.5
     - streamlit-toggle-switch >=1.0.2
+    # Temporarily diverging from upstream to fix a conflict that happens in a circular dependency.
     - htbuilder 0.6.1
     - streamlit-faker >=0.0.2
     - streamlit-card >=0.0.4
     - markdownlit >=0.0.5
     - streamlit-image-coordinates >=0.1.1,<1.0.0
+    - prometheus_client >=0.14.0
+    # upstream forgot to add validators to streamlit-extra 0.4.0, but require validators 0.22.0 for tests.
+    # Test pass with validators 0.18.2
+    - validators
```

Comment:
Should it be moved to the `test/requires` section instead?

Reply:
`validators` seems to be sitting with all the other dependencies in upstream:
https://github.com/arnaudmiribel/streamlit-extras/blob/f715952cb721f57b2841af972e40e6f3eef22f45/pyproject.toml#L36

The only odd thing is that in `0.4.0` it's missing, but during the test process it seems to still be required.
https://github.com/arnaudmiribel/streamlit-extras/blob/v0.4.0/pyproject.toml#L36

Reply:
I'll move it to the `test` section and see if it works!

Reply:
@psteyer please leave it into the runtime deps.